openedx/adg/lms/applications/analytics.py

```diff
@@ -0,0 +1,191 @@
+"""
+All the analytics related code for applications app
+"""
+from datetime import datetime
+
+from django.utils.translation import get_language_info
+from six import text_type
+
+from common.djangoapps.track import segment
+from openedx.adg.common.helpers import convert_float_point_to_percentage, get_formatted_date
+from openedx.adg.lms.applications.constants import (
+    APPLICATION_EVENTS,
+    APPLICATION_PROGRESS,
+    APPLICATION_STARTED,
+    CONTACT_INFO,
+    COURSE_COMPLETED,
+    COURSE_EVENTS,
+    COURSE_FAILED,
+    COURSE_STARTED,
+    EXPERIENCE,
+    IN_PROGRESS
+)
+
+
+def emit_application_progress(user):
+    """
+    Emits the progress for the application of the given user. The progress statuses can be Started, Submitted,
+    Waitlisted and Accepted
+
+    Arguments:
+        user (User): User to emit progress update for
+    """
+    application_progress = user.application_hub.application_progress
+
+    application_status = APPLICATION_PROGRESS.get(application_progress, APPLICATION_PROGRESS[APPLICATION_STARTED])
+    segment.identify(user.id, {'applicationStatus': application_status})
+
+    application_event = APPLICATION_EVENTS.get(application_progress, APPLICATION_EVENTS[APPLICATION_STARTED])
+    event_properties = {
+        'applicationStatus': application_status
+    }
+    if hasattr(user, 'application'):
+        event_properties['applicationId'] = user.application.id
+    segment.track(user.id, application_event, event_properties)
+
+
+def emit_user_info_update(user, application_step):
```

Comment:
Let's rename the function to indicate that this information is related to the application.

Reply:
Does `emit_user_application_related_info_update` sound better?